java/algorithm/src/main/java/org.apache.arrow.algorithm.sort/IndexSorter.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.algorithm.sort;
+
+import org.apache.arrow.vector.ValueVector;
+
+/**
+ * Sorter for the indices of a vector.
+ * @param <V> vector type.
+ */
+public class IndexSorter<V extends ValueVector> {
+
+  /**
+   * Comparator for vector indices.
+   */
+  private VectorValueComparator<V> comparator;
+
+  /**
+   * Vector indices to sort.
+   */
+  private int[] indices;
+
+  /**
+   * Sorts indices, by quick-sort. Suppose the vector is denoted by v.
```

Comment:
not for this review, but it would be nice to have  a stable sort algorithm, so we can sort multiple columns iteratively and get the resultant order.

Reply:
Good suggestion. A stable sorting algorithm helps us get consistent results. Let's leave this as a future TODO.